packages/beacon-node/src/chain/regen/regen.ts

```diff
@@ -210,13 +241,14 @@ export class StateRegenerator implements IStateRegeneratorInternal {
         });
       }
     }
+    this.modules.logger.debug("Replayed blocks to get state", {stateRoot, replaySlots});
 
     return state;
   }
 
   private findFirstStateBlock(stateRoot: RootHex): ProtoBlock {
     for (const block of this.modules.forkChoice.forwarditerateAncestorBlocks()) {
-      if (block !== undefined) {
+      if (block.stateRoot === stateRoot) {
```

Comment:
It looks like `stateRoot` wasn't used before. Did something change in `forkChoice` or this has always been a bug?

Reply:
yes this is a bug, see the PR description